src/modules/interfaces.ts

```diff
@@ -372,3 +372,95 @@ export class Module2 {
     if (!def.provides.functions) this.provides.functions = functions;
   }
 }
+
+export class ModuleBase {
+  dirname: string;
+  name: string;
+  version: string;
+  dependencies: string[];
+  provides: {
+    entities: { [key: string]: any, };
+    tables: string[];
+    functions: string[];
+    context?: Context;
+  };
+  context?: Context;
```

Comment:
What is the difference between these 2 context?

Reply:
The nested one in the `provides` object is for `src/services/iasql.ts` and maintains compatibility with the `ModuleInterface`, while the outer one is the one intended to be used by developers of modules, as all of the other `provides` fields are now auto-generated.

It simply copies the outer on to the inner one. Once the old modules are gone, we can drop the inner one without issue, but that will be in about a month.